src/bot.js

```diff
@@ -98,8 +98,26 @@ client.on("messageDelete", async deleted => {
     !modules.includes("reposting") &&
     !modules.includes("webhook")
   ) {
-    let newMessage = await deleted.channel.send(`${deleted.author || `<@${user}>`}: ${message.content || count}`);
-    gdb.set("message", newMessage.id);
+    let newMessage;
+    try {
+      let author = deleted.author;
+      if(!author) author = (await deleted.guild.members.fetch(user)).user;
+
+      let webhooks = await deleted.channel.fetchWebhooks(), webhook = webhooks.find(wh => wh.name == "Countr");
+      if (!webhook) webhook = await deleted.channel.createWebhook("Countr").catch(() => null);
+
+      if (webhook) {
+        newMessage = await webhook.send(message.content || count, {
+          username: author.username,
+          avatarURL: author.displayAvatarURL({ dynamic: true }),
+        });
+      }
+      return gdb.set("message", newMessage.id);
+
+    } catch (e) { // didn't have perms or something else went wrong
+      newMessage = await deleted.channel.send(`${deleted.author || `<@${user}>`}: ${message.content || count}`);
+      return gdb.set("message", newMessage.id);
+    }
```

Comment:
Not everyone wants it to repost as webhook. So I won't be accepting this, not now at least.

I might be able to make something different when d.js^13 comes out 😤

Reply:
You should ask though, I'm sure people will like it...

Reply:
Still, it should be optional in a way. And it might come out with d.js^13 instead then.

Reply:
I have removed it for now. But I'd highly recommend you ask because I'm very positive that everyone will like it. It's has no drawbacks...